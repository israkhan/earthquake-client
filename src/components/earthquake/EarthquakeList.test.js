/* eslint-disable no-unused-expressions */
import setupTest from "../../setupTests";
import React from "react";
import configureStore from "redux-mock-store";
import thunkMiddleware from "redux-thunk";
import enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunkMiddleware];
const mockStore = configureStore(middlewares);

import { EarthquakeList, EarthquakeListPagination, EarthquakeItem } from "../";

describe("EarthquakeList", () => {
  let store;
  beforeEach(() => {
    store = mockStore({ earthquakes: [{ id: 1 }] });
  });

  it("renders without crashing given the required props", () => {
    let wrapper = shallow(<EarthquakeList store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("does not render EarthquakeListPagination when earthquakes don't exits", () => {
    const wrapper = shallow(<EarthquakeList store={store} />);
    expect(wrapper.contains(<EarthquakeListPagination />)).toEqual(false);
  });

  it("contains the same number of EarthquakeItem components as earthquakes", () => {
    const wrapper = shallow(<EarthquakeList store={store} />);

    expect(wrapper.find(<EarthquakeItem />)).toHaveLength(0);
  });
});
