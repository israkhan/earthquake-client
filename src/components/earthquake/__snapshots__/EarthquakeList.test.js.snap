// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarthquakeList renders without crashing given the required props 1`] = `
<ContextProvider
  value={null}
>
  <EarthquakeList
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
